\documentclass[final]{beamer}

\usepackage{beamerposter}
\usetheme{confposter}

\newlength{\sepwid}
\newlength{\onecolwid}
\newlength{\twocolwid}
\newlength{\threecolwid}
\setlength{\paperwidth}{48in} % A0 width: 46.8in
\setlength{\paperheight}{36in} % A0 height: 33.1in
\setlength{\onecolwid}{0.22\paperwidth}
\setlength{\twocolwid}{0.464\paperwidth}
\setlength{\threecolwid}{0.708\paperwidth}

\usepackage{graphicx}
\usepackage{booktabs}
\usepackage{natbib}
\input{../preamble.tex}

\title{Text Modeling meets the Microbiome}
\author{Kris Sankaran and Susan P. Holmes}
\institute{Department of Statistics, Stanford University}


\begin{document}
%% \addtobeamertemplate{block end}{}{\vspace*{2ex}}
%% \addtobeamertemplate{block alerted end}{}{\vspace*{2ex}}

\begin{frame}

\begin{columns}
\begin{column}{\onecolwid}

\begin{block}{Abstract}
The human microbiome is a complex ecological system, and describing its
structure and function under different environmental conditions is important
from both basic scientific and medical perspectives. Viewed through a
biostatistical lens, many microbiome analysis goals can be approached through
latent variable modeling, for which a range of techniques are available. We
develop the analogy between text modeling, where documents are approximated as
mixtures of topics, and bacterial count modeling, where samples are approximated
as mixtures of ecological states, focusing on applications of Latent Dirichlet
Allocation, Nonnegative Matrix Factorization, and Dynamic Unigrams to the
microbiome. To develop guidelines for when different methods are appropriate, we
perform a simulation study. We further illustrate and compare techniques using
the data of \citep{dethlefsen2011incomplete}, a study on the effects of
antibiotics on bacterial community composition.
\end{block}

\begin{block}{Methods}
  \begin{itemize}


\item Latent Dirichlet Allocation: Let $x_{dv}$ be the number of times word $v$
  occurs in document $d$. Suppose the $k^{th}$ topic places weight $\beta_{vk}$
  on the $v^{th}$ term, so that $\beta_{k} \in \simplex^{V - 1}$. Suppose there
  are $N_{d}$ words in the $d^{th}$ document. Then,
\begin{align*}
x_{d\cdot} \vert \left(\beta_{k}\right)_{1}^{K} &\overset{iid}{\sim} \Mult\left(N_{d}, B\theta_{d}\right) \text{ for } d = 1, \dots, D\\
\theta_{d} &\overset{iid}{\sim} \Dir\left(\alpha\right) \text{ for } d = 1, \dots, D \\
\beta_{k} &\overset{iid}{\sim} \Dir\left(\gamma\right) \text{ for }k = 1, \dots, K,
\end{align*}
where $B = \begin{pmatrix}\beta_{1}, \dots, \beta_{K}\end{pmatrix}$.

\item Dynamic Unigram model: In light of LDA's geometric interpretation, we
  might consider in some situations a model that identifies samples with a
  continuous curve on this $V$-dimensional simplex \citep{blei2006dynamic},
\begin{align*}
x_{d\cdot} \vert \mu_{t\left(d\right)}  &\overset{iid}{\sim} \Mult\left(N_{d}, S\left(\mu_{t\left(d\right)}\right)\right) \text{ for } d = 1, \dots, D\\
\mu_{t} \vert \mu_{t - 1} &\overset{iid}{\sim} \Gsn\left(\mu_{t - 1}, \sigma^{2}I_{V}\right) \text{ for } t = 1, \dots, T \\
\mu_{0} &\overset{iid}{\sim} \Gsn\left(0, \sigma^{2}I_{V}\right),
\end{align*}
where $S$ is the multilogit link
\begin{align*}
\left[S\left(\mu\right)\right]_{v} = \frac{\exp{\mu_{v}}}{\sum_{v^{\prime}} \exp{\mu_{v^{\prime}}}},
\end{align*}
and $t\left(d\right)$ maps document $d$ to the time it was sampled.
\end{itemize}
\end{block}

\begin{figure}[!p]
  %% \centering\includegraphics[width=\textwidth]{../figure/beta_contours_v10}
  \caption{Different inference algorithms for LDA produce different uncertainty
    assessments in small sample sizes, but become comparable as more data arrives.
    Within each panel, we display the true pairs $\left(\sqrt{\beta_{v1}},
    \sqrt{\beta_{v2}}\right)$ as black labels $v$. The purple clouds are
    posterior samples from the inference procedure, which are given as row labels. The
    posterior means are labeled in orange. Different columns index different $D$
    (top column label) vs. $N$ (bottom column label) pairs. Here, we have
    subsetted to $V = 10$ -- the corresponding figure for $V = 50$ is given by
    \ref{fig:beta_contours_v50}.
  }
  \label{fig:beta_contours_v10}
\end{figure}

\begin{figure}[!p]
  \centering
  %% \includegraphics[width=\textwidth]{../figure/beta_errors_lda}
  \caption{A summary of the errors and uncertainty across models and regimes for
    the $\beta_{vk}$ in Figure \ref{fig:beta_contours_v10}. On the $x$-axis,
    we plot the difference between posterior means and the true value, after
    having square-root transformed. On the $y$-axis, we provide the standard
    deviation of the posterior samples for each $v$, along the first dimension
    $K = 1$. Columns are indexed as in Figure \ref{fig:beta_contours_v10}, but
    now rows provide different values of $V$.
    \label{fig:beta_errors_lda} }
\end{figure}
\end{column}

\begin{column}{\onecolwid}

\begin{block}{Microbiome vs. Text Analysis}
One of the primary contributions of our work is to develop the observation that
methods popular in text analysis can be adapted to the microbiome setting in a
way that produces useful summaries.

\begin{itemize}
  \item \textbf{Document} $\iff$ \textbf{Biological Sample}: The basic sampling
    units, over which conclusions are generalized, are documents (text analysis)
    and biological samples (microbiome analysis). It is of interest to highlight
    similarities and differences across these units, often through some
    variation on clustering or dimensionality reduction.
  \item \textbf{Term} $\iff$ \textbf{Bacterial species}: The fundamental
    features with which to describe samples are the counts of terms (text
    analysis) and bacterial species (microbiome analysis). More formally, by
    bacterial species, we mean Amplicon Sequence Variants
    \citep{callahan2017exact}.
  \item \textbf{Topic} $\iff$ \textbf{Community}: For interpretation, it is
    common to imagine ``prototypical'' units which can be used as a point of
    reference for observed samples. In text analysis, these are called topics --
    for example, ``business'' or ``politics'' articles have their own specific
    vocabularies. On the other hand, in microbiome analysis, these are called
    ``communities'' -- different communities have different bacterial
    signatures.
  \item \textbf{Word} $\iff$ \textbf{Sequencing Read}: A ``word'' in text analysis refers
    to a single instance of a term in a document, not its total count. The
    analog in microbiome analysis is an individual read that has been mapped to
    a unique sequence variant, though this is rarely an object of intrinsic interest.
\end{itemize}
\end{block}
\end{column}

\begin{column}{\onecolwid}
\begin{block}{ }
\begin{itemize}
  \item \textbf{Corpus} $\iff$ \textbf{Environment}: Sometimes a grouping
    structure is known apriori among sampling units \citep{teh2004sharing}. In
    the text analysis literature, a known group of documents -- for example, all
    articles coming from one newspaper -- is called a corpus. In the microbiome
    literature, the associated concept is the environment -- for example, skin,
    ocean, or soil -- from which a sample was obtained.
\end{itemize}
\end{block}

\begin{figure}[!p]
  %% \centering\includegraphics{../figure/visualize_lda_theta_boxplot-F}
  \caption{Boxplots represent approximate posteriors for estimated mixture
    memberships $\theta_{d}$, and their evolution over time. That is, each row
    of panels provides a different sequence of $\theta_{dk}$ for a single $k$,
    and different columns distinguish different phases of sampling. Note that
    the $y$-axis is on the $g$-scale, which is defined as a translated logit,
    $g\left(\mathbf{p}\right) := \left(\log p_{1} - \bar{\log \mathbf{p}},
    \dots,\log p_{K} - \bar{\log \mathbf{p}}\right)$. Note that the first
    and second antibiotic time courses result in meaningful shifts in these
    sequences, and that there appear to be long-term effects of treatment among
    bacteria in Topic 3. \label{fig:antibiotics_lda_theta}}
\end{figure}
\end{column}
\end{columns}

\end{frame}
\bibliographystyle{plainnat}
\bibliography{../refs.bib}

\end{document}
